src/pb_stub.cc

```diff
@@ -1225,6 +1172,142 @@ Stub::GetProxyStream(const int& device_id)
 #endif
 }
 
+void
+Stub::GetCUDAMemoryPoolAddress(std::unique_ptr<IPCMessage>& ipc_message)
+{
+#ifdef TRITON_ENABLE_GPU
+  bool has_exception = false;
+  std::string error_string;
+  std::unique_ptr<PbString> error_string_shm;
+
+  CUDAMemPoolMessage* cuda_pool_message_ptr = nullptr;
+  try {
+    AllocatedSharedMemory<CUDAMemPoolMessage> cuda_handle_shm =
+        shm_pool_->Load<CUDAMemPoolMessage>(ipc_message->Args());
+    cuda_pool_message_ptr = cuda_handle_shm.data_.get();
+
+    CUDAHandler& cuda_api = CUDAHandler::getInstance();
+    void* cuda_pool_address;
+    cuda_api.OpenCudaHandle(
+        device_id_, &cuda_pool_message_ptr->cuda_handle, &cuda_pool_address);
```

Comment:
While each model instance is assigned to a single GPU currently it is possible for a model instance on GPU A to return tensors on GPU B so there might be scenarios where we need to open memory pool for GPU B on model instance assigned to GPU A. In the current logic, it looks like the stub would only open the memory pool that corresponds to the `device_id_` is that correct?

Reply:
Fixed. Right now the stub can open cuda memory pool on different devices based on the device_id of the tensor.